fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/DataApiRequestImpl.java

```diff
@@ -1315,10 +1315,8 @@ protected void validateTopN(String topNRequest,  int topN, LinkedHashSet<OrderBy
                     .collect(Collectors.toMap(
                             pathSegment -> dimensionDictionary.findByApiName(pathSegment.getPath()),
                             pathSegment -> bindShowClause(pathSegment, dimensionDictionary),
-                            (LinkedHashSet<DimensionField> e, LinkedHashSet<DimensionField> i) -> {
-                                e.addAll(i);
-                                return e;
-                            },
+                            (LinkedHashSet<DimensionField> e, LinkedHashSet<DimensionField> i) ->
+                            (LinkedHashSet<DimensionField>) StreamUtils.orderedSetMerge(e, i),
```

Comment:
This should be indented eight spaces, since it's inside a lambda expression.

Also, instead of casting this to a `LinkedHashSet`, I think we should update the signature of `orderedSetMerge` to return a `LinkedHashSet`. Should be a safe change, because `LinkedHashSet` is a subclass of `Set`.

Reply:
Or perhaps made generic, so it returns the same type as the arguments.